### PR TITLE
Typing/memory block

### PR DESCRIPTION
**Pull request by armantekinalp**
Typing for rod and rigid body memory blocks.

### CODE REVIEW

File: elastica/memory_block/memory_block_rigid_body.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 
 class MemoryBlockRigidBody(RigidBodyBase, _RigidRodSymplecticStepperMixin):
-    def __init__(self, systems: Sequence, system_idx_list: Sequence[np.int64]):
+    def __init__(self, systems: Sequence, system_idx_list: Sequence[np.int64]) -> None:
```
Comment (skim0119) [inline elastica/memory_block/memory_block_rigid_body.py:10]:
```suggestion
    def __init__(self, systems: list[SystemType], system_idx_list: list[SystemIdxType]) -> None:
```

File: elastica/memory_block/memory_block_rod.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ class MemoryBlockCosseratRod(
     TODO: need more documentation!
     """
 
-    def __init__(self, systems: Sequence, system_idx_list):
+    def __init__(self, systems: Sequence, system_idx_list: tuple[int]) -> None:
```
Comment (skim0119) [inline elastica/memory_block/memory_block_rod.py:34]:
```suggestion
    def __init__(self, systems: list[RodBase], system_idx_list: list[SystemIdxType]) -> None:
```

**Comment by skim0119** [comment]
Does it pass if you run `poetry run mypy --config-file pyproject.toml elastica/memory_block`?

**Comment by armantekinalp** [comment]
> Does it pass if you run `poetry run mypy --config-file pyproject.toml elastica/memory_block`?

It doesn't because `SystemIdxType` is not defined. You have defined in PR #382 . We can wait you to merge that.

**Comment by skim0119** [comment]
`SystemIdxType` is now included in `update/mypy` branch: https://github.com/GazzolaLab/PyElastica/blob/update/mypy/elastica/typing.py

**Comment by armantekinalp** [comment]
Cool thanks, now it passes

**Comment by skim0119** [comment]
On this PR, can you also include typing for the file `elastica/modules/memory_block.py`?

**Comment by armantekinalp** [comment]
okay will work on that

**Comment by skim0119** [comment]
Also this one: https://github.com/armantekinalp/PyElastica/blob/typing/memory_block/elastica/reset_functions_for_block_structure/_reset_ghost_vector_or_scalar.py

**Review by skim0119 — APPROVED** [review]
LGTM

**Comment by skim0119** [comment]
> Also this one: https://github.com/armantekinalp/PyElastica/blob/typing/memory_block/elastica/reset_functions_for_block_structure/_reset_ghost_vector_or_scalar.py

> On this PR, can you also include typing for the file `elastica/modules/memory_block.py`?

@armantekinalp I accidently resolved them on 384

**Pull request merged**